examples/example_end_to_end_run.py

```diff
@@ -16,7 +16,12 @@
 import matplotlib.pyplot as plt
 import numpy as np
 
-# the different segments of code responsible for runing the model
+SCRIPT_PATH = os.path.abspath(os.path.dirname(__file__))
+ROOT_PATH = os.path.join(SCRIPT_PATH, "..")
+
+os.chdir(ROOT_PATH)
+
+# the different segments of code responsible for running the model
```

Comment:
This causes the file to fail pre-commit checks: `E402 Module level import not at top of file`. You have two solutions, either add the following command: `# ruff: noqa: E402` to disable the particular pre-commit check. Or move the lines below the import. I suggest the latter unless you have a good reason to keep the current ordering